dds_cli/project_status.py

```diff
@@ -102,6 +122,31 @@ def update_status(self, new_status, deadline=None, is_aborted=False, no_mail=Fal
         if is_aborted:
             extra_params["is_aborted"] = is_aborted
 
+        # If the status is going to be archived or deleted. Ask for confirmation
+        if new_status in ["Archived", "Deleted"]:
+            # get project info
+            project_info = self.get_project_info()
+
+            # Create confirmation prompt with project info
+            print_info = (
+                f"Are you sure you want to modify the status of {self.project}? All its contents "
+            )
+            if new_status == "Deleted":
+                print_info += "and metainfo "
+            print_info += (
+                "will be deleted!\n"
+                f"The project '[b]{self.project}[/b]' is about to be [b][blue]{new_status}[/blue][/b].\n"
+                f"Title: \t{project_info['Title']}\n"
+                f"Description: \t{project_info['Description']}\n"
+                f"PI: \t{project_info['PI']}\n"
+            )
+
+            dds_cli.utils.console.print(print_info)
+
+            if not rich.prompt.Confirm.ask("-"):
+                LOG.info("Probably for the best. Exiting.")
+                sys.exit(0)
+
```

Comment:
I like it, but would like to display info first and then ask for confirmation
It would be nice to have a table though. The function `show_project_info` in `project_info.py` both calls `get_project_info()` and then prints the project information as a table. It would be good if we could reuse this function, but then we'd either need to change it to a static method, or use the `ProjectInfoManager` class, or move the code to a helper function that both the `ProjectInfoManager` and `ProjectStatusManager` can use. 
   
```suggestion
        # If the status is going to be archived or deleted. Ask for confirmation
        if new_status in ["Archived", "Deleted"]:
            # Create confirmation prompt with project info
            dds_cli.utils.console.print(f"The following project '[b]{self.project}[/b]' is about to be [b][blue]{new_status}[/blue][/b].\n")
            
            # Display project info
            dds_cli.project_info.show_project_info() # needs some more because it's a public method in the ProjectInfoManager
            
            # Define question
            question = f"Are you sure you want to modify the status of this project? All its contents "
            if new_status == "Deleted":
                print_info += "and metainfo "
            print_info += "will be deleted!\n"
            
            # Ask for confirmation 
            if not rich.prompt.Confirm.ask(question):
                LOG.info("Probably for the best. Exiting.")
                sys.exit(0)
```

Reply:
@i-oden _would like to display info first and then ask for confirmation_
 What do you mean by that? It already prints the info and then asks for the user input.
 I can look at the table thing. But wouldn't it make it more difficult to test the output?

Reply:
Yeah I was a bit unclear here. At the moment it prints the following: 
```bash
Are you sure you want to modify the status of someunit00001? All its contents and metainfo will be deleted!
The project 'someunit00001' is about to be Deleted.
Title:  test
Description:    test
PI:     test@mail.com

- [y/n]: 
```
So 
1. Question
2. Info
3. Ask for input

But I'd like it to be 
1. Info
2. Question and ask for input 
----

Regarding testing - yes, the tables make it more difficult. But there are tests that check tables, e.g. `test_list_all_active_motds_table`, so check that one out for example. If testing the actual table output is too complicated and just not worth it: Skip the testing of exactly that part and only test the rest of the output etc.

Reply:
What you've done is really good, my suggestions are just some slight improvements, partly in order to reduce code reuse, and partly due to previous experience with the users. It needs to be super clear and there needs to be clear separate sections in the print outs. So
1. 'You're trying to do this'
2. <project information>
3. 'Are you sure, this will happen' 
will make it clearer for the unit users.
So in this case the tests are not the most important part, here's it's the user experience etc. The tests are a bonus but not a must.